graphql/resolvers.go

```diff
@@ -44,6 +44,80 @@ func (r *Resolver) Task() TaskResolver {
 
 type mutationResolver struct{ *Resolver }
 
+func (r *taskResolver) DependsOn(ctx context.Context, t *restModel.APITask) ([]*Dependency, error) {
+	depIds := []string{}
+	for _, dep := range t.DependsOn {
+		depIds = append(depIds, dep.TaskId)
+	}
+
+	dependencyTasks, err := task.Find(task.ByIds(depIds).WithFields(task.DisplayNameKey, task.StatusKey,
+		task.ActivatedKey, task.BuildVariantKey, task.DetailsKey, task.DependsOnKey))
+
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Cannot find dependency tasks for task %s:%s", *t.Id, err.Error()))
+	}
+
+	taskMap := map[string]*task.Task{}
+	for i := range dependencyTasks {
+		taskMap[dependencyTasks[i].Id] = &dependencyTasks[i]
+	}
+
+	task, err := task.FindOneId(*t.Id)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Unable to find task %s: %s", *t.Id, err.Error()))
+	}
+	state, err := task.BlockedState()
```

Comment:
I could be misunderstanding the intent of this, but I think there's a logic error here. If the state is blocked, it doesn't tell you anything about the status of a specific dependency - just that at least one of them is unmet and will never be met. I think the fix is to just take this out

Reply:
ah, I think this is a logic error also. I misinterpreted this line in: https://github.com/evergreen-ci/evergreen/blob/master/public/static/js/task.js#L515. 
I don't think `TaskWaiting` on `uiDep` is ever set